tests/storage/storage_migration/test_mtc_storage_class_migration.py

```diff
@@ -127,3 +133,72 @@ def test_vm_storage_class_migration_b_to_a_with_running_and_stopped_vms(
             vms_with_written_file_before_migration=written_file_to_vms_before_migration,
             target_storage_class=target_storage_class,
         )
+
+
+@pytest.mark.parametrize(
+    "vms_for_storage_class_migration",
+    [
+        pytest.param(
+            {
+                "vms_fixtures": [
+                    "vm_for_storage_class_migration_with_hotplugged_volume",
+                ]
+            },
+            id="mig_volume_hotplug_source_a_target_b",
+        )
+    ],
+    indirect=True,
+)
+class TestStorageClassMigrationWithVolumeHotplug:
+    @pytest.mark.dependency(
+        name=f"{TESTS_CLASS_NAME_VOLUME_HOTPLUG}::test_vm_storage_class_migration_with_hotplugged_volume"
+    )
+    @pytest.mark.parametrize(
+        "source_storage_class, target_storage_class, online_vms_for_storage_class_migration",
+        [
+            pytest.param(
+                {"source_storage_class": py_config[STORAGE_CLASS_A]},
+                {"target_storage_class": py_config[STORAGE_CLASS_B]},
+                {"online_vm": [True]},  # Desired VM Running status for VMs in "vms_fixtures" list
+                marks=pytest.mark.polarion("CNV-11496"),
+                id="storage_migration_a_to_b_volume_hotplug_vms",
+            )
+        ],
+        indirect=True,
+    )
+    def test_vm_storage_class_migration_with_hotplugged_volume(
+        self,
+        source_storage_class,
+        target_storage_class,
+        written_file_to_the_mounted_hotplugged_disk,
+        written_file_to_vms_before_migration,
+        online_vms_for_storage_class_migration,
+        vms_boot_time_before_storage_migration,
+        storage_mig_plan,
+        storage_mig_migration,
+        deleted_old_dvs_of_online_vms,
+    ):
+        verify_storage_migration_succeeded(
+            vms_boot_time_before_storage_migration=vms_boot_time_before_storage_migration,
+            online_vms_for_storage_class_migration=online_vms_for_storage_class_migration,
+            vms_with_written_file_before_migration=written_file_to_vms_before_migration,
+            target_storage_class=target_storage_class,
+        )
+        check_file_in_hotplugged_disk(
```

Comment:
should be in 2 separate tests so it is easy to debug and later extract coverage

Reply:
It's part of one test... ensure that both VM itself and its hotplugged disk have migrated successfully...